pkgs/by-name/sy/sysdig-cli-scanner/package.nix

```diff
@@ -2,29 +2,42 @@
   stdenv,
   lib,
   fetchurl,
-  makeWrapper,
+  writeShellScript,
 }:
 let
+  pname = "sysdig-cli-scanner";
   versionMetadata = import ./sysdig-cli-scanner.versions.nix;
   fetchForSystem = versionMetadata.${stdenv.system} or (throw "unsupported system ${stdenv.system}");
+
+  wrapper = writeShellScript "${pname}" ''
+    for arg in "$@"; do
+      # We must not pass --dbpath to the cli in case it has been called with --iac
+      # IaC Scanning does not make use of the vulnerability database
+      if [ "$arg" = "--iac" ]; then
+        exec @out@/libexec/${pname} "$@"
+      fi
+    done
+
+    # --dbpath argument is needed for vulnerability scanning mode, otherwise it tries to download
+    # the vulnerability database in the same path as the binary, which is read-only in the case of the
+    # nix store
+    exec @out@/libexec/${pname} \
+      --dbpath="$HOME/.cache/sysdig-cli-scanner/" "$@"
```

Comment:
Hey @afh, thank you for your suggestion. I started it with a grep as well to identify it `--iac` was specified, but I felt like this was a bit more flaky and could have problems if some other parameter alike would be specified. For example `--iac-something` would also match. Currently this is not a problem, and I don't think this would ever happen, but for robustness I wanted to check the actual parameters one by one with the for loop. This also avoids spawning a new command (grep) to verify if the parameter was specified.

I don't have strong opinion on this, so I would be fine with your change. If it works, it works. 

The usage of ${pname} within the /libexec/ path ensures for me that I am not making any typo, but I agree to change it to plain text instead of using the variable.

Reply:
Thanks for taking a look, @tembleking, I had hoped that the `-w` flag on grep would prevent `--iac` from matching on `--iac-something` unfortunately it does not ☹️ 

I see your point and spawning an extra command, vs. checking the options in "pure" shell. Please keep it as it is now, my suggestion can be viewed as somewhat of a premature optimization that almost optimizes one thing but at too great of a cost on another.